govcd/org_test.go

```diff
@@ -174,6 +175,119 @@ func (vcd *TestVCD) Test_Admin_GetVdcByName(check *C) {
+		check.Skip(fmt.Sprintf("VDC '%s' already exists", TestCreateOrgVdc))
+	}
+
+	results, err := vcd.client.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":   "providerVdc",
+		"filter": fmt.Sprintf("(name==%s)", vcd.config.VCD.ProviderVdc.Name),
+	})
+	check.Assert(err, IsNil)
+	if len(results.Results.VMWProviderVdcRecord) == 0 {
+		check.Skip(fmt.Sprintf("No Provider VDC found with name '%s'", vcd.config.VCD.ProviderVdc.Name))
+	}
+	providerVdcHref := results.Results.VMWProviderVdcRecord[0].HREF
+
+	results, err = vcd.client.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":   "providerVdcStorageProfile",
+		"filter": fmt.Sprintf("(name==%s)", vcd.config.VCD.ProviderVdc.StorageProfile),
+	})
+	check.Assert(err, IsNil)
+	if len(results.Results.ProviderVdcStorageProfileRecord) == 0 {
+		check.Skip(fmt.Sprintf("No storage profile found with name '%s'", vcd.config.VCD.ProviderVdc.StorageProfile))
+	}
+	providerVdcStorageProfileHref := results.Results.ProviderVdcStorageProfileRecord[0].HREF
+
+	results, err = vcd.client.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":   "networkPool",
+		"filter": fmt.Sprintf("(name==%s)", vcd.config.VCD.ProviderVdc.NetworkPool),
+	})
+	check.Assert(err, IsNil)
+	if len(results.Results.NetworkPoolRecord) == 0 {
+		check.Skip(fmt.Sprintf("No network pool found with name '%s'", vcd.config.VCD.ProviderVdc.NetworkPool))
+	}
+	networkPoolHref := results.Results.NetworkPoolRecord[0].HREF
+
+	allocationModels := []string{"AllocationVApp", "AllocationPool", "ReservationPool"}
+	for i, allocationModel := range allocationModels {
+		vdcConfiguration := &types.VdcConfiguration{
+			Name:            fmt.Sprintf("%s%d", TestCreateOrgVdc, i),
+			Xmlns:           "http://www.vmware.com/vcloud/v1.5",
+			AllocationModel: allocationModel,
+			ComputeCapacity: []*types.ComputeCapacity{
+				&types.ComputeCapacity{
+					CPU: &types.CapacityWithUsage{
+						Units:     "MHz",
+						Allocated: 1024,
+						Limit:     1024,
+					},
+					Memory: &types.CapacityWithUsage{
+						Allocated: 1024,
+						Limit:     1024,
+					},
+				},
+			},
+			VdcStorageProfile: &types.VdcStorageProfile{
+				Enabled: true,
+				Units:   "MB",
+				Limit:   1024,
+				Default: true,
+				ProviderVdcStorageProfile: &types.Reference{
+					HREF: providerVdcStorageProfileHref,
+				},
+			},
+			NetworkPoolReference: &types.Reference{
+				HREF: networkPoolHref,
+			},
+			ProviderVdcReference: &types.Reference{
+				HREF: providerVdcHref,
+			},
+			IsEnabled:            true,
+			IsThinProvision:      true,
+			UsesFastProvisioning: true,
+		}
+		task, err := adminOrg.CreateVdc(vdcConfiguration)
+		check.Assert(task, Equals, Task{})
+		check.Assert(err, Not(IsNil))
+		check.Assert(err.Error(), Equals, "VdcConfiguration missing required field: ComputeCapacity[0].Memory.Units")
+		vdcConfiguration.ComputeCapacity[0].Memory.Units = "MB"
+
+		err = adminOrg.CreateVdcWait(vdcConfiguration)
+		check.Assert(err, IsNil)
+
```

Comment:
Thanks for your updates. We're almost there. A few requests for the creation test:
1. As mentioned above, the test must check whether the VDC already exists before creating it. As it is, if the VDC exists, we get an ugly error message. Given that the test creates a VDC that is supposed to be deleted soon, I think a safe course of action would be to delete it and then proceed with the creation. 
2. The addition to the cleanup list works fine, but it should not be the main method for deletion. It is there as a safety net to take over when the test function exits prematurely. Instead, we should have an explicit deletion, and the test for its success (`err` equals nil, and a search for the VDC returns an empty result)

Reply:
Dang sorry about that, I wrote and tested the check for the VDC already existing before adding all the allocation models.  I moved it inside the loop and ran the tests with and without the VDCs already existing.

Gotcha, I misinterpreted what you meant and have added the call to `vdc.DeleteWait` back into the test.  I've also added a refresh on the org after deletion to ensure an empty result.

Thank you for your patience!